core/state_processor.go

```diff
@@ -157,6 +167,13 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 				fdb.LastAccHash = accIt.Hash()
 				fdb.LastSlotHash = stIt.Hash()
 			}
+			log.Info("Collected and prepared key values from base tree", "count", count, "duration", time.Since(now))
+
+			now = time.Now()
+			if err := mkv.migrateCollectedKeyValues(tt.Overlay()); err != nil {
+				return nil, nil, 0, fmt.Errorf("could not migrate key values: %w", err)
+			}
```

Comment:
After we finish collecting them, we let the component do the go-verkle use of new APIS: batch create leaf nodes, and insert into the tree. More about this below.